app/app.js

```diff
@@ -137,30 +139,44 @@ window.run = function() {
             });
         };
 
-        var appLayoutPromise = Application.getOSInformation().then(function(osInfo) {
-            if (BaseConfig.useTabLayout) {
-                return createTabBarLayout(counterBadgeControllerPromise);
-            } else {
-                return createDrawerLayout(counterBadgeControllerPromise);
-            }
-        });
+        var createLayout = function() {
+            return BaseConfig.useTabLayout
+                ? createTabBarLayout()
+                : createDrawerLayout();
+        };
 
-        // Show welcome modal only after layout is created for proper
-        // bookkeeping of the active view.
-        Promise.join(
-            appLayoutPromise,
-            welcomeModalControllerPromise,
-        function(menuController, welcomeModalController) {
-            // The welcome modal can be configured to show only
-            // once -- on initial startup, by passing in the
-            // parameter `{forced: false}` below
-            welcomeModalController.show({forced: true});
-
-            // Deep linking services will enable deep linking on startup
-            // and while running
-            // It will open the deep link in the current active tab
-            deepLinkingServices = new DeepLinkingServices(menuController);
-        });
+        var initMainLayout = function() {
+            return createLayout().then(function(menuController) {
+                // Deep linking services will enable deep linking on startup
+                // and while running it will open the deep link in the current
+                // active tab
+                deepLinkingServices = new DeepLinkingServices(menuController);
+            });
+        };
+
+        var runApp = function() {
+            welcomeModalControllerPromise.then(function(welcomeModalController) {
+                // The welcome modal can be configured to show only once
+                // (on first launch) by setting `{forced: false}` as the
+                // parameter for welcomeModalController.show()
+                welcomeModalController
+                    .show({forced: true})
+                    .finally(initMainLayout);
+            });
+        };
+
+        var runAppPreview = function() {
+            MobifyPreviewPlugin.init()
+            .then(function(previewPlugin) {
+                previewPlugin
+                    .preview(BaseConfig.baseURL, BaseConfig.previewBundle)
+                    .then(runApp);
+            });
+        };
+
+        // Configure the previewEnabled flag located in baseConfig.js
+        // to enable/disable app preview
+        BaseConfig.previewEnabled ? runAppPreview() : runApp();
```

Comment:
@jvoll  What about having a default "dev" config that we require in that's checked into git and have BuddyBuild overwrite that if we want? BuddyBuild supports setting environment variables which we could use to trigger writing the file.